packages/react-query/src/internals/getArrayQueryKey.ts

```diff
@@ -5,6 +5,11 @@ export type QueryKey = [
   { input?: unknown; type?: Exclude<QueryType, 'any'> }?,
 ];
 
+export type QueryKeyKnown<TInput, TType extends Exclude<QueryType, 'any'>> = [
+  string[],
+  { input: TInput; type: TType },
+];
```

Comment:
I had to add this as `QueryKey` is not generic and I wasn't able to make it generic as it considers the second array item to be optional. Changes on it are welcome!